### PR TITLE
[mb-6434] Orders document viewer benchmark network device conditions

### PR DESCRIPTION
**Pull request by duncan-truss**
## Description

This PR builds upon the work that Donald began in https://github.com/transcom/mymove/pull/5886 to benchmark the loading of the largest file for the TOO orders document viewer.

This PR sets up configuration so we can throttle emulating the network at various speeds fast (10Mb/s), medium (5Mb/s), and slow (1Mb/s) outlined in the Jira story.  I've also added the ability to slowdown the cpu rate to simulate running on older hardware.

It also tries to generalize some of the settings for global configuration and command line arguments to set us up to more easily add tests and debug.

## Reviewer Notes

Please double check my math and speed throttling units in https://github.com/transcom/mymove/blob/008b26592be47cb99046405c57e124129787c38d/puppeteer/constants.js#L41

I had to suppress a number of eslint rules that will get flagged by Danger.js

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

Uncomment `export STORAGE_BACKEND=s3`from .envrc

```sh
direnv allow
aws-vault exec transcom-gov-dev -- make db_bandwidth_up
aws-vault exec transcom-gov-dev -- make server_run
make client_run
node puppeteer/benchmarking.js -m network-comparison
```

If you want to see it run in the browser you can set `headless: false` in the puppeteer/config.json file.

Also you can run `node puppeteer/benchmarking.js run --help` to see some of the command usage we've added.

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6434) for this change
* [Benchmark tools research](https://docs.google.com/document/d/1qF8fLZOKUAbDXGiMwWF4J2OjRzaQG9wm54CivZjpqv8/edit#heading=h.iqfo3yj7xhna)

## Screenshots

Screen Shot 2021-02-12 at 2 57 33 PM

### CODE REVIEW

**Review by donaldthai — APPROVED** [review]
This looks really good to me! 🚀

File: puppeteer/constants.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      upload: { type: 'number' },
+      latency: { type: 'number' },
+    },
+  },
+  cpuSlowdownRate: { type: 'number' },
+};
+
+// Chrome devtools expects throughput in bytes https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/lib/emulation.js#L70
+// Puppeteer only has 2 network presets https://github.com/puppeteer/puppeteer/blob/main/src/common/NetworkConditions.ts#L27
+const networkProfiles = {
```
Comment (donaldthai) [inline puppeteer/constants.js:41]: One thing I do wonder is if it's more realistic to setup the connection based on simulated connection types (cable, dsl, 3g connections, etc.)? There's some profiles from https://github.com/WPO-Foundation/webpagetest/blob/master/www/settings/connectivity.ini.sample that we could possibly include to expand our testing later.

Comment (monfresh) [reply puppeteer/constants.js:41]: From what I've found online, 1 Mbps (megabit, not megabyte) is 125,000 bytes per second. As long as we're only changing the first digit, the relative difference stays the same, so I think this is probably fine.

**Comment by monfresh** [comment]
This is great! As a future enhancement, it would be nice to be able to set the headless mode via the CLI.

**Comment by duncan-truss** [comment]
> This is great! As a future enhancement, it would be nice to be able to set the headless mode via the CLI.

I thought of making it a command line arg but I liked having all of the puppeteer options in config.json.

**Pull request merged**